include/zephyr/posix/sys/utsname.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_POSIX_SYS_UTSNAME_H_
+#define ZEPHYR_INCLUDE_POSIX_SYS_UTSNAME_H_
+
+#include <errno.h>
+#include <zephyr/toolchain/common.h>
+#include <zephyr/net/hostname.h>
+#include "version.h"
```

Comment:
Just a minor nit - I can't remember which RFC it was but we should preemptively adopt @gmarull 's suggestion for canonical header ordering.

```suggestion
#include "version.h"

#include <errno.h>

#include <zephyr/net/hostname.h>
#include <zephyr/toolchain/common.h>
```